meetings/2021-07/july-14.md

```diff
@@ -0,0 +1,1028 @@
+
+DDC: So moving on, next up are decorator modifiers. So like I said before, modifiers are the ability to prepend a keyword to the Decorator itself in order to give an additional capability. And this was really the way that we figured out how to solve a common use case, which is the ability to add initializers to class elements and classes themselves. Actually initializers be at in a keyword adds the ability to add initializers to the elements and those initializers run for any.
+
+ I'm sorry to interrupted, I think Mark had a question about previous slide. And since you're taking questions in the middle, I might just do the queue now. I know Jordans on the queue as well.
+
+DDC: That's fine. I'm okay with that answering questions.
+
+MM: Yeah, so when you do the decorated private field, does the private field still provide the actual storage for the private field value, or do the get and set provide - I think I'm a little confused here. The definer of the decorator can provide a getter and setter for the private field that provides the storage instead of the private field?
+
+DDC: So, the getter and setter can be called with an instance, you have to do (?) call and instance and they they are functions that can the private slot on the instance and set the private slot on the instance.
+
+MM: Okay? So the privates, lot of the instance still exists and if it's still where the private value is stored,
+
+DDC: yes, it does not to change the shape of the class in any way and it doesn't change the way the class works at all. If you don't use the access object or those functions, then nothing changes at all. It's like they were never there. Using them is just a way to access the values in a separate way.
+
+MM: And if the getter and setter is applied to anything other than an instance, it throws?
+
+DDC: it should. If that's not in the spec then I should definitely add it. Any other questions?
+
+JHD: Yeah, it was a couple slides back where you're showing the decorator function interface. Yeah, here you go. I was just curious if the context access set function, can that ever fail, or it was that only there on a private field and so like it should never fail.
+
+DDC: If it is called on an instance of the class, because the class should always have that, it should never fail.
+
+JHD: It should never fail, but if it's called on the instead something else, so it'll either throw because the field isn't on the receiver. The receiver or it will succeed. Is that correct?
```

Comment:
```suggestion
JHD: It should never fail, but if it's called instead on something else, so it'll either throw because the field isn't on the receiver, or it will succeed. Is that correct?
```